src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/poptimizer/costmodel/CFRNumberOfTermsShippedInRequests.java

```diff
@@ -51,6 +51,16 @@ else if ( lop instanceof LogicalOpBGPAdd ) {
 
 			futureIntResSize = initiateCardinalityEstimation(subplan);
 		}
+		else if ( lop instanceof LogicalOpGPAdd ) {
+			final LogicalOpGPAdd gpAdd = (LogicalOpGPAdd) lop;
+			numberOfTerms = QueryPatternUtils.getNumberOfTermOccurrences( gpAdd.getPattern() );
+
+			final PhysicalPlan subplan = plan.getSubPlan(0);
+			final PhysicalPlan reqGP = PhysicalPlanFactory.extractRequestAsPlan(gpAdd);
+			numberOfJoinVars = ExpectedVariablesUtils.intersectionOfCertainVariables(subplan,reqGP).size();
+
+			futureIntResSize = initiateCardinalityEstimation(subplan);
+		}
```

Comment:
After updating the helper function 'extractRequestAsPlan' in the class PhysicalPlanFactory, the implementation from line 34 to line 63 can be updated as:
```
if ( lop instanceof UnaryLogicalOp ) {
	SPARQLGraphPattern pattern = null;
	if (lop instanceof LogicalOpTPAdd) {
		pattern = ((LogicalOpTPAdd) lop).getTP();

	} else if (lop instanceof LogicalOpBGPAdd) {
		pattern = ((LogicalOpBGPAdd) lop).getBGP();

	} else if (lop instanceof LogicalOpGPAdd) {
		pattern = ((LogicalOpGPAdd) lop).getPattern();
	}

	numberOfTerms = QueryPatternUtils.getNumberOfTermOccurrences( pattern );

	final PhysicalPlan subplan = plan.getSubPlan(0);
	final PhysicalPlan req = PhysicalPlanFactory.extractRequestAsPlan( (UnaryLogicalOp) lop );
	numberOfJoinVars = ExpectedVariablesUtils.intersectionOfCertainVariables(subplan, req).size();

	futureIntResSize = initiateCardinalityEstimation(subplan);
}
```